src/devices/interlogix.c

```diff
@@ -0,0 +1,392 @@
+ * 54 F5 negative latch bit
+ * 55 even parity over odd bits 15-55
+ * 56 odd parity over even bits 16-56
+ * 57 zero/one, programmable
+ * 58 RF on for 366 uS (old stop bit)
+ * 59 one
+ * 60-62 modulus 8 count of number of ones in bits 15-54
+ * 63 zero (new stop bit)
+ *
+ * Protocol Description
+ * ________________________________
+ * Bits 00 to 02 are a 976 ms RF front porch pulse, providing a wake up period that allows the
+ *      system controller receiver to synchronize with the incoming packet.
+ * Bits 3 to 14 include 12 sync pulses, e.g., logical 0's, to synchronize the receiver.
+ * Bit 15 is a start pulse, e.g., a logical 1, that tells the receiver that data is to follow.
+ * Bits 16-58 provide information regarding the transmitter and associated sensor. In other
+ *      embodiments, bits 16-58 may be replaced by an analog signal.
+ * Bits 16 to 35 provide a 20-bit sensor identification code that uniquely identifies the particular
+ *      sensor sending the message. Bits 36 to 39 provide a 4 bit device-type code that identifies the
+ *      specific-type of sensor, e.g., smoke, PIR, door, window, etc. The combination of the sensor
+ *      bits and device bits provide a set of data bits.
+ * Bits 40 through 42 provide a 3-bit trigger count that is incremented for each group of message
+ *      packets. The trigger count is a simple but effective way for preventing a third party from
+ *      recording a message packet transmission and then re-transmitting that message packet
+ *      transmission to make the system controller think that a valid message packet is being transmitted.
+ * Bit 43 provides the low battery bit.
+ * Bits 44 through 53 provide the latch bit value and the debounced value for each of the five inputs
+ *      associated with the transmitter. For the F5 input, both a positive and negative latch bit are provided.
+ * Bit 55 provides even parity over odd bits 15 to 55.
+ * Bit 56 provides odd parity over even bits 16 to 56.
+ * Bit 57 is a programmable bit that can be used for a variety of applications, including providing an
+ *      additional bit that could be used for the sensor identification code or device type code.
+ * Bit 58 is a 366 ms RF on signal that functions as the "old" stop bit. This bit provides compatibility with
+ *      prior system controllers that may be programmed to receive a 58-bit message.
+ * Bit 59 is a logical 1.
+ * Bits 60 to 62 are a modulus eight count of the number of 1 bits in bits 15 through 54, providing enhanced
+ *      error detection information to be used by the system controller. Finally, bit 63 is the "new" stop bit,
+ *      e.g., a logical 0, that tells the system controller that it is the end of the message packet.
+ *
+ *
+ *
+ */
+
+#define INTERLOGIX_CRC_POLY    0x07
+#define INTERLOGIX_CRC_INIT    0x00
+
+//preamble (and single sync bit) should be 0000 0000 0000 1
+//TODO: only searching for 0000 0001 and should be searching for all 13 bits of preamble
```

Comment:
a preamble isn't meant to be matched completly, the sender or receiver could be "warming up" and the signal distorted. You can safely write `only searching for bottom 8 bits of the 13 bits preamble`